dimod/roof_duality/fix_variables.py

```diff
@@ -0,0 +1,68 @@
+from dimod.roof_duality._fix_variables import fix_variables_wrapper
+from dimod.vartypes import Vartype
+
+
+def fix_variables(bqm, sampling_mode=True):
+    """Determine assignments for some bqm variables.
+
+    fix_variables uses maximum flow in the implication network to correctly fix variables (that is,
+    one can find an assignment for the other variables that attains the optimal value). The
+    variables that roof duality fixes will take the same values in all optimal solutions.
+
+    Args:
+        bqm (:obj:`.BinaryQuadraticModel`)
+            A binary quadratic model.
+
+        sampling_mode (bool, optional, default=True):
+            Setting sampling_model to False can fix more variables, but in some optimal solutions
+            these variables may take different values.
+
+    Returns:
+        dict: Variable assignments for some bqm variables.
+
+    Examples:
+
+        >>> bqm = dimod.BinaryQuadraticModel.empty(dimod.SPIN)
+        >>> bqm.add_variable('a', 1.0)
+        >>> dimod.fix_variables(bqm)
+        {'a': -1}
+
+        There are two ground states, so no variables fixed.
```

Comment:
In the next example, there

(Otherwise looks like it's for the previous example.)